### ISSUE TITLE
bug: yield is not an unused expression

### ISSUE THREAD

**Issue by remitbri** [eslint/eslint#1672]
ESLint 0.12.0
with .eslintrc
```js
{
  "ecmaFeatures" : {
    "generators": true
  }
}
```
If `no-unused-expressions` is enabled, the following is considered an error

```js
function *generator() {
    yield "44";
    yield "55";
}
```

From what the newbie me sees, it seems it would be just a matter of modifying line 63 of lib/rules/no-unused-expressions.js to

```js
!/^(?:Assignment|Call|New|Update|Yield)Expression$/.test(type) &&
```

**Comment by michaelficarra** [comment]
Yeah, that should do it.

**Comment by remitbri** [comment]
How should I proceed from there? I'm really new to proper testing, and if I update tests/lib/rules/no-unused-expressions.js to add `"yield someValue"` as a valid test, `npm test`fails ("Unexpected token yield") ; it seems odd to PR the rule fix only without the test also.

**Comment by michaelficarra** [comment]
Since `yield` isn't allowed in ES5, you'll need to enable ES6 features in the parser for those tests with `ecmaFeatures: { "generators": true }`. See other tests for examples.

**Comment by remitbri** [comment]
Thanks. 
That I did,

```js
        {
            code: "yield \"2\"",
            ecmaFeatures: { "generators": true }
        },
```
… but now, depending on what I add to the valid section (`yield "2"`, `yield someValue`), I get an error `'Unexpected string'` / `'Unexpected identifier'`

**Comment by michaelficarra** [comment]
Try using it in a generator. `(function*(){ yield 0; })`.

**Comment by remitbri** [comment]
Thanks for your patient help. 
Actually the anonymous generator didn't work (`Unexpected token (`), I had to use `function* foo(){ yield 0; }`

**Comment by michaelficarra** [comment]
That's because the generator was in statement position. You could wrap the generator in parentheses to avoid the name if you wanted.

**Comment by remitbri** [comment]
Hmm. It doesn't work ("`Expected an assignment or function call and instead saw an expression.`"), unless I use `(function*(){ yield 0; })()`. I think it "looks" simpler if I just use a name.
